lib/services/services.c

```diff
@@ -404,6 +404,28 @@ services_alert_create(const char *id, const char *exec, int timeout,
     return action;
 }
 
+/*!
+ * \brief Set the user and group that an action will execute as
+ *
+ * \param[in,out] action  Action to modify
+ * \param[in]     user    Name of user to execute action as
+ * \param[in]     group   Name of group to execute action as
+ *
+ * \return pcmk_ok on success, -errno otherwise
+ *
+ * \note This will have no effect unless the process executing the action runs
+ *       as root, and the action is not a systemd or upstart action.
+ *       We could implement this for systemd by adding User= and Group= to
+ *       [Service] in the override file, but that seems more likely to cause
+ *       problems than be useful.
```

Comment:
Alternatively with systemd, actions could be triggered with the equivalent
of `systemctl --user`, combined with existing changing effective user/group.